### PR TITLE
Bugfixesv2

### PR DESCRIPTION
**Pull request by manu-unbxd**
bug fixes from harsh

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,7 +206,9 @@ facet configuration table is available here.
 | isSearchable | Boolean | if you wish to have a search feature on facets |
 | searchPlaceHolder | String | placeholder for the facet search input |
 | textFacetWrapper | String | css class for the facets list |
-
+| viewMore | Booelan | For enabling view more / less functionality for individual facets  |
```
Comment (Anupama-Unbxd) [inline README.md:209]: lets call this `enableViewMore` to indicate its a boolean value

File: README.md [hunk 2]
```diff
@@ -213,7 +206,9 @@ facet configuration table is available here.
...
+| viewMoreTxt | Array | The text to show in button eg:  ["show all", "show less"] |
```
Comment (Anupama-Unbxd) [inline README.md:210]: lets not use short texts for the config names, can this be renamed to `viewMoreText` ?

File: demo/js/index.js [hunk 3]
```diff
@@ -86,15 +183,18 @@ window.unbxdSearch.updateConfig({
...
+    onFacetLoad:function(facets){
+    },
+    isCollapsible:true,
+    isSearchable:true,
+    viewMore:true,
   },
   pagination: {
     type:'FIXED_PAGINATION',
-    el:document.querySelector(".unxPagination"),
+    el:document.querySelector("#paginationContainer"),
```
Comment (Anupama-Unbxd) [inline demo/js/index.js:197]: is this a class exposed by the SDK? if yes then it should begin with the prefix `UNX-`

Comment (manu-unbxd) [reply demo/js/index.js:197]: no.. it will be  DOM element from customer

File: public/examples/internetExplorer.html [hunk 4]
```diff
@@ -102,7 +103,7 @@
     </script>
     <script src="../dist/js/iePolyfill.js"></script>
 
-     <script src="https://libraries.unbxdapi.com/search-sdk/v0.1.4/vanillaSearch.js"></script>
+     <script src="../dist/js/vanillaSearch.js"></script>
```
Comment (Anupama-Unbxd) [inline public/examples/internetExplorer.html:106]: Do not add local links in example files as customers will refer to this

File: README.md (hunk 1 above)
Comment (manu-unbxd) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (manu-unbxd) [reply README.md]: done

File: public/examples/internetExplorer.html (hunk 4 above)
Comment (manu-unbxd) [reply public/examples/internetExplorer.html:106]: removed

File: src/core/createLayout.js [hunk 5]
```diff
@@ -67,34 +61,27 @@ const createLayout = function() {
...
     this.options.facet.multiLevelFacetTemplate.bind(this);
     this.options.facet.facetTemplate.bind(this);
     this.options.facet.facetItemTemplate.bind(this);
+    if(spellCheck.el) {
+        spellCheck.el.innerHTML = ``;
+        spellCheck.el.appendChild(this.spellCheckWrapper);
+    }
+    if(this.options.facet.el) {
+        this.options.facet.el.innerHTML = ``;
+        this.options.facet.el.appendChild(this.facetsWrapper);
```
Comment (Anupama-Unbxd) [inline src/core/createLayout.js:79]: store `this.options.facet.el` in a variable and use it

File: src/common/options.js [hunk 6]
```diff
@@ -208,15 +209,28 @@ const options = {
...
-                                `data-swatch-target=".UNX-img-block" class="${this.swatchClass}" style="background-color:${item}"> </button>`].join('')
-                }
-            });
-            return `<div class="UNX-swatch-color-list">${btnUI}</div>`;
+            let btnList = ``;
+            let imgsUI = ``;
+            const {
+                swatchClass
+            } = this.options.swatches;
+            if(swatchImgs.length > 1) {
```
Comment (Anupama-Unbxd) [inline src/common/options.js:216]: disable swatches by default as all customers will not have it

File: styles/index.scss [hunk 7]
```diff
@@ -625,6 +638,46 @@ button.UNX-page-next {
...
 }
+.UNX-facets-all {
+  transition: all .3s ease-in;
+}
+.UNX-accordian-close .UNX-facets-all {
+  max-height: 0px;
+  overflow: hidden;
+  opacity: 0;
+}
+.UNX-facets-item.UNX-view-more {
```
Comment (Anupama-Unbxd) [inline styles/index.scss:649]: can we use just one class i.e. `.UNX-view-more` ?

File: styles/index.scss [hunk 8]
```diff
@@ -614,7 +627,7 @@ button.UNX-page-next {
   font-size: 12px;
   opacity: .5;
 }
-.UNX-facet-close:before {
+button.UNX-facet-close:before {
```
Comment (Anupama-Unbxd) [inline styles/index.scss:630]: why is `button` tag required here? can we avoid the over specification

File: styles/index.scss [hunk 9]
```diff
@@ -137,12 +145,17 @@ body {
...
 }
 .UNX-swatch-btn {
-  border-color: transparent;
+  border: solid 1px transparent;
   margin: 5px 2px;
   height: 24px;
   width: 24px;
   box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.43);
 }
+.UNX-swatch-btn.UNX-selected-swatch,
```
Comment (Anupama-Unbxd) [inline styles/index.scss:154]: can we just use one class here to avoid over specification

File: styles/index.scss [hunk 10]
```diff
@@ -586,7 +599,7 @@ button.UNX-page-next {
   font-weight: 300;
   line-height: 32px;
 }
-.UNX-facet-open:before {
+button.UNX-facet-open:before {
```
Comment (Anupama-Unbxd) [inline styles/index.scss:602]: can we remove the `button` tag here?

File: styles/index.scss [hunk 11]
```diff
@@ -625,6 +638,46 @@ button.UNX-page-next {
...
+.view-more-row {
+  display: flex;
+  justify-content: flex-end;
+}
+.view-more-row button {
+  background: transparent;
+  border-color: transparent;
+  color: #071fd8;
+}
+.range-facets-block .UNX-facets-item {
```
Comment (Anupama-Unbxd) [inline styles/index.scss:670]: any reason `view-more-row` & `range-facets-block` classes are not prefixed with `UNX-` ?

File: styles/index.scss [hunk 12]
```diff
@@ -625,6 +638,46 @@ button.UNX-page-next {
...
+.UNX-accordian-open .UNX-facets-all {
+  max-height: 300px;
+  opacity: 1;
+  overflow: scroll;
+}
+.view-more-row {
+  display: flex;
+  justify-content: flex-end;
+}
+.view-more-row button {
```
Comment (Anupama-Unbxd) [inline styles/index.scss:665]: can we give a class to the button & use that instead? this gives us the flexibility to change the HTML tag from button to anything else later without updating the CSS

File: src/core/createLayout.js (hunk 5 above)
Comment (manu-unbxd) [reply src/core/createLayout.js:79]: removed.

File: styles/index.scss (hunk 12 above)
Comment (manu-unbxd) [reply styles/index.scss]: done

**Pull request merged**